mm2src/coins/lp_coins.rs

```diff
@@ -1293,16 +1297,26 @@ pub struct SendTakerFundingArgs<'a> {
     pub trading_amount: BigDecimal,
     /// Unique data of specific swap
     pub swap_unique_data: &'a [u8],
+    /// As of now, this field is specifically used to wait for confirmations of ERC20 approval transaction.
+    pub wait_for_confirmation_until: u64,
 }
 
 /// Helper struct wrapping arguments for [TakerCoinSwapOpsV2::refund_taker_funding_secret]
 pub struct RefundFundingSecretArgs<'a, Coin: ParseCoinAssocTypes + ?Sized> {
     pub funding_tx: &'a Coin::Tx,
-    pub time_lock: u64,
+    pub funding_time_lock: u64,
+    pub payment_time_lock: u64,
     pub maker_pubkey: &'a Coin::Pubkey,
     pub taker_secret: &'a [u8],
     pub taker_secret_hash: &'a [u8],
+    pub maker_secret_hash: &'a [u8],
     pub swap_contract_address: &'a Option<BytesJson>,
+    /// DEX fee
+    pub dex_fee: &'a DexFee,
+    /// Additional reward for maker (premium)
+    pub premium_amount: BigDecimal,
+    /// Actual volume of taker's payment
+    pub trading_amount: BigDecimal,
```

Comment:
Why do we need these?

Reply:
these args are needed for Eth taker refund methods
(payment_amount = trading_amount + premium_amount)

https://github.com/KomodoPlatform/etomic-swap/blob/5e15641cbf41766cd5b37b4d71842c270773f788/contracts/EtomicSwapTakerV2.sol#L208

https://github.com/KomodoPlatform/etomic-swap/blob/5e15641cbf41766cd5b37b4d71842c270773f788/contracts/EtomicSwapTakerV2.sol#L267